tests/beta/tapes/test_cv_param_shift.py

```diff
@@ -0,0 +1,829 @@
+# Copyright 2018-2020 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Unit tests for the CV parameter-shift CVParamShiftTape"""
+import pytest
+import numpy as np
+
+import pennylane as qml
+from pennylane.beta.tapes import CVParamShiftTape, qnode
+from pennylane.beta.queuing import expval, var, sample, probs, MeasurementProcess
+
+
+hbar = 2
+
+
+class TestGradMethod:
+    """Tests for parameter gradient methods"""
+
+    def test_non_differentiable(self):
+        """Test that a non-differentiable parameter is
+        correctly marked"""
+
+        with CVParamShiftTape() as tape:
+            qml.FockState(1, wires=0)
+            qml.Displacement(0.543, 0, wires=[1])
+            qml.Beamsplitter(0, 0, wires=[0, 1])
+            expval(qml.X(wires=[0]))
+
+        assert tape._grad_method(0) is None
+        assert tape._grad_method(1) == "A"
+        assert tape._grad_method(2) == "A"
+        assert tape._grad_method(3) == "A"
+        assert tape._grad_method(4) == "A"
+
+        tape._update_gradient_info()
+
+        assert tape._par_info[0]["grad_method"] is None
+        assert tape._par_info[1]["grad_method"] == "A"
+        assert tape._par_info[2]["grad_method"] == "A"
+        assert tape._par_info[3]["grad_method"] == "A"
+        assert tape._par_info[4]["grad_method"] == "A"
+
+    def test_no_graph_exception(self):
+        """Test that an exception is raised for analytically differentiable
+        operations if use_graph=False"""
+        with CVParamShiftTape() as tape:
+            qml.Rotation(0.543, wires=[0])
+            expval(qml.P(0))
+
+        with pytest.raises(ValueError, match="must always use the graph"):
+            tape._grad_method(0, use_graph=False)
+
+    def test_independent(self):
+        """Test that an independent variable is properly marked
+        as having a zero gradient"""
+
+        with CVParamShiftTape() as tape:
+            qml.Rotation(0.543, wires=[0])
+            qml.Rotation(-0.654, wires=[1])
+            expval(qml.P(0))
+
+        assert tape._grad_method(0) == "A"
+        assert tape._grad_method(1) == "0"
+
+        tape._update_gradient_info()
+
+        assert tape._par_info[0]["grad_method"] == "A"
+        assert tape._par_info[1]["grad_method"] == "0"
+
+    def test_finite_diff(self, monkeypatch):
+        """If an op has grad_method=F, this should be respected
+        by the CVParamShiftTape"""

```

Comment:
Based on this docstring, would have thought it's an example for a fallback to finite differences, although it doesn't seem to be the case

Reply:
It should be! Here, I am monkeypatching the rotation option to only specify `grad_method="F"`. The CV tape should respect this, and return a grad method of "F" for that particular gate argument.

Reply:
Note that one F parameter does not cause the entire circuit to require finite-difference---only that parameter gradient is computed using finite diff, the rest continue to use the parameter shift rule.